server/conn.go

```diff
@@ -1297,6 +1298,15 @@ func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool
 	req := rs.NewRecordBatch()
 	gotColumnInfo := false
 	for {
+		failpoint.Inject("SleepInwriteChunks", func(val failpoint.Value) {
+			time.Sleep(time.Duration(val.(int)) * time.Millisecond)
+		})
+		//MaxExecDuration() return 0 if there is no limit
+		if rs.MaxExecDuration().Nanoseconds() > 0 &&
+			time.Now().After(rs.StartExecTime().Add(rs.MaxExecDuration())) {
+			return errors.New("Query execution was interrupted, max_execution_time exceeded")
```

Comment:
But this cannot work for cases like `HashAgg` which consumes all of its child's input and finally returns result?

Reply:
It's better to start a new `goroutine` to monitor all processing queries periodically and kill these who exceed their `max_execution_time` like #10350.

Reply:
It's better to start a new goroutine to monitor all processing queries periodically and kill these who exceed their max_execution_time like #10350.